src/v0/destinations/salesforce/utils.js

```diff
@@ -140,4 +156,30 @@ const getAccessToken = async (destination) => {
   });
 };
 
-module.exports = { getAccessToken, salesforceResponseHandler };
+const collectAuthorizationInfo = async (event) => {
+  let authorizationFlow;
+  let authorizationData;
+  if (isDefinedAndNotNull(event.metadata?.secret)) {
```

Comment:
Is there no corner scenario where `event.metadata?.secret` might come as null for oauth destination ?
Instead why not make use of destDef to have clear segregation ? Or have salesforce_oauth/transform.js just as wrapper class setting `authorizationFlow=OAUTH` & re-use the existing code ?